087-adopt-connect-health-endpoint.md

```diff
@@ -0,0 +1,84 @@
+
+This is the default HTTP probe configuration shared by all components:
+
+| Property name       | Default value | Description                                                                                      |
+|---------------------|---------------|--------------------------------------------------------------------------------------------------|
+| initialDelaySeconds | 60            | The initial delay before first the health is first checked.                                      |
+| timeoutSeconds      | 5             | The timeout for each attempted health check.                                                     |
+| periodSeconds       | 5             | The timeout for each attempted health check.                                                     |
+| successThreshold    | 1             | Minimum consecutive successes for the probe to be considered successful after having failed.     |
+| failureThreshold    | 3             | Minimum consecutive failures for the probe to be considered failed after having succeeded.       |
+
+Strimzi does not support the HTTPs protocol for Kafka Connect REST API ([KIP-208](https://cwiki.apache.org/confluence/display/KAFKA/KIP-208%3A+Add+SSL+support+to+Kafka+Connect+REST+interface)).
+
+Example output with `/` endpoint:
+
+```sh
+$ kubectl exec my-cluster-kafka-0 -- curl -s http://my-connect-cluster-connect-api:8083 | jq
+{
+  "version": "3.9.0",
+  "commit": "a60e31147e6b01ee",
+  "kafka_cluster_id": "_xIIeVIOQMKXimv_l96WtQ"
+}
+```
+
+## Motivation
+
+Using the `/` endpoint for Kafka Connect and Mirror Maker 2 health checks is a common approach, but it does not actually test for readiness, because these requests can be completed before the worker is started.
+Instead, the `/health` endpoint waits for the worker startup phase to complete, which is made of internal topics creation if they do not exists, internal topics full read, and cluster join.
+
+If the worker has not yet completed the startup phase, or it is unable to respond in time, the response will have a 5xx status code.
+All Kafka Connect endpoints have a timeout of 90 seconds, with the exception of the `/health` endpoint where it is hardcoded to 10 seconds for faster unhealty worker detection.
+Unlike the `/` endpoint, the `/health` endpoint response message includes error details that can help with troubleshooting.
+
+Example output with `/health` endpoint:
+
+```sh
+$ kubectl exec my-cluster-kafka-0 -- curl -s http://my-connect-cluster-connect-api:8083/health | jq
+{
+  "status": "healthy",
+  "message": "Worker has completed startup and is ready to handle requests."
+}
+```
+
+## Proposal
+
+Strimzi will support the Kafka 3.8 branch for an extended period of time ([SIP-77](https://github.com/strimzi/proposals/blob/main/077-support-for-kafka-4.0.md#extended-support-for-kafka-39)) due to the upcoming major Kafka release.
```

Comment:
Removing the extended support part (my bad), I still think that it would be good to provide `/health` endpoint support as early as possible. The change to get rid of the 3.8 endpoint fallback logic is very simple, and there will be unit tests failing as a reminder. Wdyt?